task/generate-olm-catalog/0.1/generate-olm-catalog.yaml

```diff
@@ -0,0 +1,61 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: generate-olm-catalog
+spec:
+  description: Generate an OLM catalog index and the Dockerfile to build
+  params:
+    - name: binary-image
+      description: Image in which to build catalog
+      default: registry.redhat.io/openshift4/ose-operator-registry:latest
+    - name: bundle-image
+      description: Comma separated list of bundles to add
+    - name: operator-name
+      description: Name of the Operator
+    - name: operator-version
+      description: Version of the Operator
+    - name: default-channel
+      description: The channel that subscriptions will default to if unspecified
+      default: stable
+  workspaces:
+    - name: source
+      description: Workspace with the source code
+  steps:
+    - name: generate-olm-catalog
+      image: "registry.redhat.io/openshift4/ose-operator-registry:latest"
+      workingDir: $(workspaces.source.path)/source
+      securityContext:
+        runAsUser: 0
+        capabilities:
+          add:
+            - SETFCAP
+      script: |
+        #!/usr/bin/env bash
+
+        set -xe
+
+        mkdir -p catalog/
+
+        opm generate dockerfile catalog/ --binary-image=$(params.binary-image)
+
+        opm init $(params.operator-name) \
+          --default-channel=$(params.default-channel) \
+          --description=./README.md \
+          --output=yaml > catalog/operator.yaml
+
+        opm render $(params.bundle-image) --output=yaml >> catalog/operator.yaml
+
+        cat << EOF >> catalog/operator.yaml
```

Comment:
I don't think that this will work by default like this. Not with the latest image at least. The latest versions of opm do not generate an output that is compatible with older catalogs due to the version of the opm CLI.